src/frontend/apps/drive/src/features/ui/preview/not-supported/NotSupportedPreview.tsx

```diff
@@ -11,6 +11,18 @@ interface NotSupportedPreviewProps {
 
 export const NotSupportedPreview = ({ file }: NotSupportedPreviewProps) => {
   const { t } = useTranslation();
+
+  const handleDownload = async () => {
+    // Temporary solution, waiting for a proper download_url attribute.
+    const a = document.createElement("a");
+    a.style.display = "none";
+    a.href = file.url!;
+    a.download = file.title;
+    document.body.appendChild(a);
+    a.click();
+    document.body.removeChild(a);
+  };
+
```

Comment:
The logic is still duplicated between `ExplorerGridActionsCell` `FilePreview` `NotSupportedPreview`, make all use only one function nope?